### PR TITLE
fix(builders): fix vulnerability in webpack-dev-server

### LINKED ISSUE #971: npm audit shows webpack-dev-server  as  vulnerably
_[Please make sure you have read the submission guidelines before posting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-issue)_

# Prerequisites
 

## Expected Behavior

Please describe the behavior you are expecting

## Current Behavior

What is the current behavior?

```
 npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Missing Origin Validation                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.1.11                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @angular-devkit/build-angular [dev]                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @angular-devkit/build-angular > webpack-dev-server           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/725                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Missing Origin Validation                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.1.11                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @nrwl/builders [dev]                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @nrwl/builders > webpack-dev-server                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/725                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 2 high severity vulnerabilities in 134732 scanned packages
  2 vulnerabilities require manual review. See the full report for details.

```
 
### Context

  npx nx --version
7.2.0-beta.1

### PR DESCRIPTION
**Pull request by FrozenPandaz**
## TL;DR:

The request origin of the HMR websocket server was unchecked, allowing any remote server to access the source code, thus possibly stealing it.

Related Security Advisory: https://www.npmjs.com/advisories/725

## Current Behavior

Workspaces have a vulnerability in `webpack-dev-server` through `@nrwl/builders` and `@angular-devkit/build-angular`

## Expected Behavior

Workspaces do not have vulnerabilities

This fixes the vulnerability through `@nrwl/builders`. We can provide a migration after there's a new version of `@angular-devkit/build-angular`.

Related PR: https://github.com/angular/angular-cli/pull/13347

## Issue

Partially Fixes #971

### CODE REVIEW

**Review by Toxicable — CHANGES_REQUESTED** [review]
Could you note the reasoning for bumping this dep in the commit message?

**Comment by FrozenPandaz** [comment]
@Toxicable Good idea. Done!

**Review by Toxicable — APPROVED** [review]
